File: calculadora/calculadora/src/main/java/com/example/calculadora/entities/HistorialEntity.java
package com.example.calculadora.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Controller;

import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name="historial")
public class HistorialEntity {
    @Id
    @Column(nullable=false, unique=true)
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long  id;

    @Column(nullable=false, unique=true)
    private Integer posicion;

    private String descripcion;
    private String ecuacion;
    private String resultado;
    private LocalDateTime fecha;
}

File: calculadora/calculadora/src/main/kotlin/com/example/calculadora/entities/HistorialEntity.kt
package com.example.calculadora.entities

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "historial", schema= "calculadora")
open class HistorialEntity(

        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        @Column(nullable = false, unique = true)
        open var id: Long? = null,
        open var userId: Long? = null,

        open var descripcion: String? = null,
        open var ecuacion: String? = null,
        open var resultado: String? = null,

        @Column(nullable = false)
        open var fecha: LocalDateTime? = null
        )